### PR TITLE
feat(card): New full screen enter button design

### LINKED ISSUE #771: Full screen card expand icon isn't styled properly in dark mode
```r
page_fillable("data-bs-theme" = "dark", card(full_screen = TRUE, "Card body"))
```

Screenshot 2023-08-29 at 5 21 40 PM

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes #771
Supersedes #772 

- Color utilities now communicate color via `--bslib-color-fg` and `--bslib-color-bg` CSS variables
- The full screen icon takes the card background and foreground colors
- I decided not to use a drop shadow since it added too much complication around getting those colors right
- More CSS variables in the card full screen enter class to make things a bit more customizable

## Examples

image

image

## TODO

- [x] `value_box_theme(fg = , bg = )` set those `--bslib-color-fg` and `--bslib-color-bg` variables too.
- [x] Set box shadow to `rgba(0, 0, 0, 0.15)` always
- [x] Set initial hover opacity to `0.7`

### CODE REVIEW

File: inst/bslib-scss/color-utilities.scss [hunk 1]
```diff
@@ -23,12 +23,28 @@ $bslib-enable-color-utilities: $bootstrap-version >= 5 !default;
...
+
+  // Fill in the `--color-*` variables
+  @each $name, $color in $theme-colors {
+    .text-#{$name} {
+      --color-fg: #{$color};
+    }
+    .bg-#{$name} {
+      $contrast: color-contrast($color);
+      --color-bg: #{$color};
+      --color-fg: #{$contrast};
```
Comment (cpsievert) [inline inst/bslib-scss/color-utilities.scss:47]:
```suggestion
      --color-bg: #{$color};
      --color-fg: #{color-contrast($color)};
```

File: inst/bslib-scss/color-utilities.scss [hunk 2]
```diff
@@ -23,12 +23,28 @@ $bslib-enable-color-utilities: $bootstrap-version >= 5 !default;
   // Named color background and foreground utility classes ---------------------
   @each $name, $color in $bslib-gradient-colors {
     .bg-#{$name} {
-      background-color: $color;
-      color: color-contrast($color);
+      $contrast: color-contrast($color);
+      --color-bg: #{$color};
+      --color-fg: #{$contrast};
```
Comment (cpsievert) [inline inst/bslib-scss/color-utilities.scss:28]:
```suggestion
      --bslib-color-bg: #{$color};
      --bslib-color-fg: #{$contrast};
```

File: R/card.R [hunk 3]
```diff
@@ -300,13 +300,11 @@ card_init_js <- function() {
...
+  # https://www.visiwig.com/icons/
+  # https://www.visiwig.com/icons-license/
+  HTML('<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="bi bi-arrows-angle-expand " style="height:1em;width:1em;fill:currentColor;" aria-hidden="true" role="img"><path d="M20 5C20 4.4 19.6 4 19 4H13C12.4 4 12 3.6 12 3C12 2.4 12.4 2 13 2H21C21.6 2 22 2.4 22 3V11C22 11.6 21.6 12 21 12C20.4 12 20 11.6 20 11V5ZM4 19C4 19.6 4.4 20 5 20H11C11.6 20 12 20.4 12 21C12 21.6 11.6 22 11 22H3C2.4 22 2 21.6 2 21V13C2 12.4 2.4 12 3 12C3.6 12 4 12.4 4 13V19Z"/></svg>')
```
Comment (cpsievert) [inline R/card.R:305]: Do we need the classes here?

**Review by cpsievert — APPROVED** [review]

File: R/card.R (hunk 3 above)
Comment (gadenbuie) [reply R/card.R:305]: They don't do anything unless you've loaded the bootstrap icons font icon styles... so no? I don't think we're relying on the class and I don't think we should even if we were, so I'll take them out.

**Pull request merged**